### ISSUE TITLE
Packer with ansible provisioners (Windows 2016 AMI creation on AWS) - Failed to connect to the host via ssh: Warning

### ISSUE THREAD

**Issue by HCLCDR** [hashicorp/packer#8726]
- Packer Version:  1.5.1
builder: amazon-ebs
os_type: Windows
communicator: winrm

- Ansible Version: 2.9.3
- Python Version: 2.7.5
   
### Debug Logs:

==> amazon-ebs: Provisioning with Ansible...
2020/02/16 19:09:14 packer-provisioner-ansible plugin: SSH proxy: serving on 127.0.0.1:34071
==> amazon-ebs: Executing Ansible: ansible-playbook --extra-vars packer_build_name=amazon-
ebs packer_builder_type=amazon-ebs -o IdentitiesOnly=yes -i /tmp/packer-provisioner-ansible805584169 /root/win-playbook1.yml -e ansible_ssh_private_key_file=/tmp/ansible-key475469426 --connection packer -vvv --extra-vars ansible_shell_type=powershell ansible_shell_executable=None
    amazon-ebs: ansible-playbook 2.9.3
    amazon-ebs:   config file = /etc/ansible/ansible.cfg
    amazon-ebs:   configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
    amazon-ebs:   ansible python module location = /usr/lib/python2.7/site-packages/ansible
    amazon-ebs:   executable location = /bin/ansible-playbook
    amazon-ebs:   python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
    amazon-ebs: Using /etc/ansible/ansible.cfg as config file
    amazon-ebs: host_list declined parsing /tmp/packer-provisioner-ansible805584169 as it did not pass its verify_file() method
    amazon-ebs: script declined parsing /tmp/packer-provisioner-ansible805584169 as it did not pass its verify_file() method
    amazon-ebs: auto declined parsing /tmp/packer-provisioner-ansible805584169 as it did not pass its verify_file() method
    amazon-ebs: Parsed /tmp/packer-provisioner-ansible805584169 inventory source with ini plugin
    amazon-ebs:
    amazon-ebs: PLAYBOOK: win-playbook1.yml ****************************************************
    amazon-ebs: 1 plays in /root/win-playbook1.yml
    amazon-ebs:
    amazon-ebs: PLAY [all] *********************************************************************
    amazon-ebs: META: ran handlers
    amazon-ebs:
    amazon-ebs: TASK [18.9.26.1.1,CCE-37775-4 | Ensure 'Application - Control Event Log behavior when the log file reaches its maximum size' is set to 'Disabled'] ***
    amazon-ebs: task path: /root/win-playbook1.yml:6
    amazon-ebs: Using module file /usr/lib/python2.7/site-packages/ansible/modules/windows/win_regedit.ps1
    amazon-ebs: Pipelining is enabled.
    amazon-ebs: <127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: root
    amazon-ebs: <127.0.0.1> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=34071 -o 'IdentityFile="/tmp/ansible-key475469426"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/187f5d0b62 127.0.0.1 'chcp.com 65001 > $null ; PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UABvAHcAZQByAFMAaABlAGwAbAAgAC0ATgBvAFAAcgBvAGYAaQBsAGUAIAAtAE4AbwBuAEkAbgB0AGUAcgBhAGMAdABpAHYAZQAgAC0ARQB4AGUAYwB1AHQAaQBvAG4AUABvAGwAaQBjAHkAIABGwAQQBHAE0AQQBYAHcAQgAzAEEASABJAEEAWQBRAEIAdwBBAEgAQQBBAFoAUQBCAHkAQQBBAD0APQA='
2020/02/16 19:09:14 packer-provisioner-ansible plugin: SSH proxy: accepted connection
2020/02/16 19:09:15 packer-provisioner-ansible plugin: authentication attempt from 127.0.0.1:57412 to 127.0.0.1:34071 as root using none
2020/02/16 19:09:15 packer-provisioner-ansible plugin: authentication attempt from 127.0.0.1:57412 to 127.0.0.1:34071 as root using publickey
2020/02/16 19:09:15 packer-provisioner-ansible plugin: new env request: LANG=en_US.UTF-8
2020/02/16 19:09:15 packer-provisioner-ansible plugin: new exec request: chcp.com 65001 > $null ; PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UABvAHcAZQByAFMAaABlAGwAbAAgAC0ATgBvAFAAcgBvAGYAaQBsAGUAIAAtAE4AbwBuAEkAbgB0AGUAcgBhAGMAdABpAHYAZQAgAC0ARQB4AGUAYwB1AHQAaQBvAG4AUABvAGwAaQBjAHkAIABGwAQQBHAE0AQQBYAHcAQgAzAEEASABJAEEAWQBRAEIAdwBBAEgAQQBBAFoAUQBCAHkAQQBBAD0APQA=
2020/02/16 19:09:15 packer-provisioner-ansible plugin: [INFO] 112272 bytes written for 'stdin'
2020/02/16 19:09:15 [INFO] 112272 bytes written for 'stdin'
2020/02/16 19:09:15 packer-builder-amazon-ebs plugin: [INFO] starting remote command: chcp.com 65001 > $null ; PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UABvAHcAZQByAFMAaABlAGwAbAAgAC0ATgBvAFAAcgBvAGYAaQBsAGUAIAAtAE4AbwBuAEkAbgB0AGUAcgBhAGMAdABpAHYAZQAgAC0ARQB4AGUAYwB1AHQAaQBvAG4AUABvAGwAaQBjAHkAIABGwAQQBHAE0AQQBYAHcAQgAzAEEASABJAEEAWQBRAEIAdwBBAEgAQQBBAFoAUQBCAHkAQQBBAD0APQA=
2020/02/16 19:09:15 packer-builder-amazon-ebs plugin: [INFO] command 'chcp.com 65001 > $null ; PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UABvAHcAZQByAFMAaABlAGwAbAAgAC0ATgBvAFAAcgBvAGYAaQBsAGUAIAAtAE4AbwBuAEkAbgB0AGUAcgBhAGMAdABpAHYAZQAgAC0ARQB4AGUAYwB1AHQAaQBvAG4AUABvAGwAaQBjAHkAIABGwAQQBHAE0AQQBYAHcAQgAzAEEASABJAEEAWQBRAEIAdwBBAEgAQQBBAFoAUQBCAHkAQQBBAD0APQA=' exited with code: 1
2020/02/16 19:09:15 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 1
2020/02/16 19:09:15 [INFO] 0 bytes written for 'stdout'
2020/02/16 19:09:15 [INFO] 34 bytes written for 'stderr'
2020/02/16 19:09:15 [INFO] RPC client: Communicator ended with: 1
2020/02/16 19:09:15 [INFO] RPC endpoint: Communicator ended with: 1
2020/02/16 19:09:15 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdout'
2020/02/16 19:09:15 packer-provisioner-ansible plugin: [INFO] 34 bytes written for 'stderr'
2020/02/16 19:09:15 packer-provisioner-ansible plugin: [INFO] RPC client: Communicator ended with: 1
    amazon-ebs: <127.0.0.1> (1, '', "Warning: Permanently added '[127.0.0.1]:34071' (RSA) to the list of known hosts.\r\nParameter format not correct - ;\r\n")
    amazon-ebs: <127.0.0.1> Failed to connect to the host via ssh: Warning: Permanently added '[127.0.0.1]:34071' (RSA) to the list of known hosts.
    amazon-ebs: Parameter format not correct - ;
    amazon-ebs: fatal: [default]: FAILED! => {
    amazon-ebs:     "changed": false,
    amazon-ebs:     "module_stderr": "Warning: Permanently added '[127.0.0.1]:34071' (RSA) to the list of known hosts.\r\nParameter format not correct - ;\r\n",
    amazon-ebs:     "module_stdout": "",
    amazon-ebs:     "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    amazon-ebs:     "rc": 1
    amazon-ebs: }
    amazon-ebs:
    amazon-ebs: PLAY RECAP *********************************************************************
    amazon-ebs: default                    : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    amazon-ebs:
2020/02/16 19:09:15 packer-provisioner-ansible plugin: shutting down the SSH proxy
2020/02/16 19:09:15 [INFO] (telemetry) ending ansible
==> amazon-ebs: Provisioning step had errors: Running the cleanup provisioner, if present...
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary keypair...
2020/02/16 19:09:45 [INFO] (telemetry) ending amazon-ebs
2020/02/16 19:09:45 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/02/16 19:09:45 machine readable: amazon-ebs,error []string{"Error executing Ansible: Non-zero exit status: exit status 2"}
==> Builds finished but no artifacts were created.
2020/02/16 19:09:45 [INFO] (telemetry) Finalizing.
Build 'amazon-ebs' errored: Error executing Ansible: Non-zero exit status: exit status 2

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Error executing Ansible: Non-zero exit status: exit status 2

==> Builds finished but no artifacts were created.
2020/02/16 19:09:46 waiting for all plugin processes to complete...
2020/02/16 19:09:46 /usr/local/bin/packer: plugin process exited
2020/02/16 19:09:46 /usr/local/bin/packer: plugin process exited
2020/02/16 19:09:46 /usr/local/bin/packer: plugin process exited

#### Overview of the Issue########

I am using Packer 1.5.1, ansible 2.9.3 to build Windows 2016 AMI in AWS environment (Public AMI ID ami-08c7081300f7d9abb)
Connected to WinRM, able to run powershell-provisioner, however not able to run ansible provisioner for hardening Windows 2016 based on CIS benchmark

Error:  amazon-ebs: <127.0.0.1> Failed to connect to the host via ssh: Warning: Permanently added

#### Reproduction Steps####

Steps to reproduce this issue

### Packer version###

1.5.1

### Simplified Packer Buildfile [winpacker1]####

```
  "builders": [{
    "type":"amazon-ebs",
    "access_key": "{{user `aws_access_key`}}",
    "secret_key": "{{user `aws_secret_key`}}",
    "region": "{{user `aws_region`}}",
    "instance_type": "t2.medium",
    "iam_instance_profile": "Admin",
    "associate_public_ip_address": false,

    "availability_zone":"us-east-1b",
    "vpc_id":"{{user `aws_vpc_id`}}",
    "subnet_id":"{{user `aws_subnet_id`}}",
    "security_group_id":"{{user `aws_security_group_id`}}",
    "ami_name": "packer-example-{{timestamp}}",
    "user_data_file": "windows-aws1.txt",
    "communicator": "winrm",
    "winrm_username": "Administrator",
    "winrm_password": "SuperS3cr3t!",

    "source_ami_filter": {
      "filters": {
        "virtualization-type": "hvm",
        "name": "Windows_Server-2016-English-Full-Base-2019.11.13",
        "root-device-type": "ebs"
      },
      "owners": ["amazon"],
      "most_recent": true
    },
    "launch_block_device_mappings": [
    {
      "device_name": "/dev/xvda",
      "volume_size": 80,
      "volume_type": "gp2",
      "delete_on_termination": true
    }
  ]
  }],

  "provisioners": [{
    "type": "powershell",
    "inline": [
      "dir c:\\"
    ]
  },
  {
      "type":  "ansible",
      "playbook_file": "./win-playbook1.yml",
      "extra_arguments": [
        "--connection", "packer", "-vvv",
        "--extra-vars", "ansible_shell_type=powershell ansible_shell_executable=None"
      ]
  }]
}
```

### User_data_file [windows-aws1.txt]###

<powershell>
##Set administrator password
net user Administrator SuperS3cr3t!
wmic useraccount where "name='Administrator'" set PasswordExpires=FALSE

##First, make sure WinRM can't be connected to
netsh advfirewall firewall set rule name="Windows Remote Management (HTTP-In)" new enable=yes action=block

##Delete any existing WinRM listeners
winrm delete winrm/config/listener?Address=*+Transport=HTTP  2>$Null
winrm delete winrm/config/listener?Address=*+Transport=HTTPS 2>$Null

##Create a new WinRM listener and configure
winrm create winrm/config/listener?Address=*+Transport=HTTP
winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="0"}'
winrm set winrm/config '@{MaxTimeoutms="7200000"}'
winrm set winrm/config/service '@{AllowUnencrypted="true"}'
winrm set winrm/config/service '@{MaxConcurrentOperationsPerUser="12000"}'
winrm set winrm/config/service/auth '@{Basic="true"}'
winrm set winrm/config/client/auth '@{Basic="true"}'

##Configure UAC to allow privilege elevation in remote shells
$Key = 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System'
$Setting = 'LocalAccountTokenFilterPolicy'
Set-ItemProperty -Path $Key -Name $Setting -Value 1 -Force

##Configure and restart the WinRM Service; Enable the required firewall exception
Stop-Service -Name WinRM
Set-Service -Name WinRM -StartupType Automatic
netsh advfirewall firewall set rule name="Windows Remote Management (HTTP-In)" new action=allow localip=any remoteip=any
Start-Service -Name WinRM
</powershell>

### Sample YML file "win-playbook1.yml" I'm using - for reference###

```
---
- hosts: localhost
  gather_facts: no

  tasks:
   - name: "18.9.26.1.1,CCE-37775-4 | Ensure 'Application - Control Event Log behavior when the log file reaches its maximum size' is set to 'Disabled'"
     win_regedit:
        path: HKLM:\Software\Policies\Microsoft\Windows\Eventlog\Application
        name: "Retention"
        data: "0"
        type: string

   - name: "18.9.8.2,CCE-38217-6 | Ensure 'Set the default behavior for AutoRun' is set to 'Enabled - Do not execute any autorun commands'"
     win_regedit:
        path: HKLM:\Software\Microsoft\Windows\Currentversion\Policies\Explorer
        name: "NoAutorun"
        data: "1"
        type: dword

```

### Operating system and Environment details###

Windows_Server-2016-English-Full-Base-2019.11.13

### Connection_plugins, OS version and packer-log file attached###

I'm using the following connection_plugins: https://github.com/hashicorp/packer/tree/master/examples/ansible/connection-plugin/2.6.x/packer.py

OS version of the Build Machine: CentOS Linux release 7.5.1804 (Core)

### Log Fragments and crash.log files###

Packer-log file captured from the build machine::  [Packer-log.txt](https://github.com/hashicorp/packer/files/4211426/Packer-log.txt)

**Comment by HCLCDR** [comment]
Additional Comments and packer-log file attached:

**I'm using the following connection_plugins**:  https://github.com/hashicorp/packer/tree/master/examples/ansible/connection-plugin/2.6.x/packer.py

OS version of the Build  Machine:  CentOS Linux release 7.5.1804 (Core)

**Please find attached packer-log file captured from the build machine**::
[packer-log951268096.docx](https://github.com/hashicorp/packer/files/4204909/packer-log951268096.docx)

**Comment by HCLCDR** [comment]
###Connection_plugins, OS version and packer-log file attached###

I'm using the following connection_plugins: https://github.com/hashicorp/packer/tree/master/examples/ansible/connection-plugin/2.6.x/packer.py

OS version of the Build Machine: CentOS Linux release 7.5.1804 (Core)

Packer-log file captured from the build machine:
[Packer-log.txt](https://github.com/hashicorp/packer/files/4210732/Packer-log.txt)

**Comment by HCLCDR** [comment]
###Connection_plugins, OS version and packer-log file attached###

I'm using the following connection_plugins: https://github.com/hashicorp/packer/tree/master/examples/ansible/connection-plugin/2.6.x/packer.py

OS version of the Build Machine: CentOS Linux release 7.5.1804 (Core)

Packer-log file captured from the build machine:
[Packer-log.txt](https://github.com/hashicorp/packer/files/4210743/Packer-log.txt)

**Comment by jonestk** [comment]
Are there any fixes for this? I'm having the same issue within Azure.

**Comment by SwampDragons** [comment]
Sorry for the late reply; I think this is probably going to be solved by #8625, which allows you to disable the localhost proxy which has been a bit of a hack workaround for getting Packer to play nicely with Ansible up until now. If you follow the link to that Pr, you'll find links to test builds, and notes on how to disable the proxy in the PR's documentation updates.
